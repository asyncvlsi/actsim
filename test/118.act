deftype struct (int a, b) { }

function foo (int x) : struct
{ 
  chp {
    self.a := x;
    self.b := ~x
  }
}

defproc test()
{
  int y, z;

  chp {
    y := foo (3).a;
    z := foo (3).b;
    log ("%bgot ", y, "%b and ", z)
  }
}
