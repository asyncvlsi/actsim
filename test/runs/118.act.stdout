[                  40] <>  got 00000000000000000000000000000011 and 11111111111111111111111111111100
